graylog2-web-interface/src/components/indices/IndexMaintenanceStrategiesConfiguration.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+
+import { Select } from 'components/common';
+
+const IndexMaintenanceStrategiesConfiguration = React.createClass({
+  propTypes: {
+    title: React.PropTypes.string.isRequired,
+    description: React.PropTypes.string.isRequired,
+    selectPlaceholder: React.PropTypes.string.isRequired,
+    pluginExports: React.PropTypes.array.isRequired,
+    strategies: React.PropTypes.array.isRequired,
+    activeConfig: React.PropTypes.object.isRequired,
+    updateState: React.PropTypes.func.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      activeStrategy: this.props.activeConfig.strategy,
+      activeConfig: this.props.activeConfig.config,
+      newStrategy: this.props.activeConfig.strategy,
+      newConfig: this.props.activeConfig.config,
+    };
+  },
+
+  _getDefaultStrategyConfig(selectedStrategy) {
+    const result = this.props.strategies.find((strategy) => strategy.type === selectedStrategy);
+    return result ? result.default_config : undefined;
+  },
+
+  _getStrategyJsonSchema(selectedStrategy) {
+    const result = this.props.strategies.find((strategy) => strategy.type === selectedStrategy);
+    return result ? result.json_schema : undefined;
+  },
+
+  _getStrategyConfig(selectedStrategy) {
+    if (this.state.activeStrategy === selectedStrategy) {
+      // If the newly selected strategy is the current active strategy, we use the active configuration.
+      return this.state.activeConfig;
+    } else {
+      // If the newly selected strategy is not the current active strategy, we use the selected strategy's default config.
+      return this._getDefaultStrategyConfig(selectedStrategy);
+    }
+  },
+
+  _onSelect(newStrategy) {
+    if (!newStrategy || newStrategy.length < 1) {
+      this.setState({newStrategy: undefined});
+      return;
+    }
+
+    const newConfig = this._getStrategyConfig(newStrategy);
+
+    this.setState({newStrategy: newStrategy, newConfig: newConfig});
+    this.props.updateState(newStrategy, newConfig);
+  },
+
+  _onConfigUpdate(newConfig) {
+    this.setState({newConfig: newConfig});
+    this.props.updateState(this.state.newStrategy, newConfig);
+  },
+
+  _availableSelectOptions() {
+    return this.props.pluginExports.map((config) => {
+      return {value: config.type, label: config.displayName}
+    });
+  },
+
+  _getConfigurationComponent(selectedStrategy) {
+    if (!selectedStrategy || selectedStrategy.length < 1) {
+      return;
+    }
+
+    const strategy = this.props.pluginExports.find((strategy) => strategy.type === selectedStrategy);
+
+    if (!strategy) {
+      return undefined;
```

Comment:
It is okay (from my perspective) to do it in `getInitialState` because it signals that this variable is not initialized. I guess it is only done in older components, because it is in 99% of the cases omittable, actually. It helps when debugging/logging to see if an expression is `null` or `undefined`, because for the first case it is very probable that it was returned on purpose.

For the evaluation of an equality check it does not matter, yes.